syft/pkg/cataloger/common/generic_cataloger.go

```diff
@@ -56,18 +56,44 @@ func (c *GenericCataloger) Catalog(resolver source.FileResolver) ([]pkg.Package,
 			continue
 		}
 
+		pkgsForRemoval := make(map[artifact.ID]artifact.Identifiable)
```

Comment:
nit: Since you're never accessing the "value" in the key-value pair (i.e. the `artifact.Identifiable` value here), you don't need to store an `artifact.Identifiable` in the map. You can just use a map type of `map[artifact.ID]struct{}`, and store `struct{}{}` values to indicate that a given `artifact.ID` is present in the set. This uses less memory, but is also simpler to follow, since it makes clear that the map item's value is never used, other than to determine its existence.

Reply:
Interesting, I like that. I found [benchmark agreeing with you](https://gist.github.com/davecheney/3be245c92b61e5045f75).